cmd/main.go

```diff
@@ -163,24 +163,27 @@ func main() {
 		}
 		frrInstance := frr.NewFRR(ctx, reloadStatus, logger, logging.Level(logLevel))
 
-		if err = (&controller.FRRConfigurationReconciler{
-			Client:     mgr.GetClient(),
-			Scheme:     mgr.GetScheme(),
-			FRRHandler: frrInstance,
-			Logger:     logger,
-			NodeName:   nodeName,
-		}).SetupWithManager(mgr); err != nil {
+		configReconciler := &controller.FRRConfigurationReconciler{
+			Client:       mgr.GetClient(),
+			Scheme:       mgr.GetScheme(),
+			FRRHandler:   frrInstance,
+			Logger:       logger,
+			NodeName:     nodeName,
+			ReloadStatus: reloadStatus,
+		}
+		if err = configReconciler.SetupWithManager(mgr); err != nil {
 			setupLog.Error(err, "unable to create controller", "controller", "FRRConfiguration")
 			os.Exit(1)
 		}
 
 		if err = (&controller.FRRStateReconciler{
-			Client:    mgr.GetClient(),
-			Scheme:    mgr.GetScheme(),
-			FRRStatus: frrInstance,
-			Logger:    logger,
-			NodeName:  nodeName,
-			Update:    reloadStatusChan,
+			Client:           mgr.GetClient(),
+			Scheme:           mgr.GetScheme(),
+			FRRStatus:        frrInstance,
+			Logger:           logger,
+			NodeName:         nodeName,
+			Update:           reloadStatusChan,
+			ConversionResult: configReconciler,
```

Comment:
nit/general thought: the controllers are all in the same pkg, is there a reason not too have these as vars on the pkg scope? I feel it is more clunky this way (main caring that a particular controller is a "conversionFetcher" and should be passed to the other) rather than just letting the relation happen on the pkg. wdyt?

Reply:
If I got the comment right...

Using global variables is a shortcut that removes any form of encapsulation. It's like putting pipes between the otherwise self contained objects. 

The `ConversionResult` field is an external dependency that this controller has with something that lives outside of it, which _incidentally_ is another controller living in the same package. As an external dependency, it must come from outside and injected to the object.

This way is more modular and sets a clear boundary between the various objects.  The extra cost, as you said, is the wiring / injection of the various dependencies but I think this way is cleaner.